File: ServerSpace/src/main/java/com/api/server/EntListaController.java
package com.api.server;

import Business.EntLista;
import EntityLayer.EntListaEntity;
import Models.ResponseAPI;
import Models.EntLista.EntListaModel;

import java.util.ArrayList;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.PathVariable;
import org.springframework.web.bind.annotation.RestController;
import org.springframework.web.bind.annotation.RequestMapping;

@RestController
@RequestMapping("/api/EntLista")
public class EntListaController extends Configuracion {

    @GetMapping("/ObtenerItemsCodigo/{codigo}")
    public ResponseAPI<ArrayList<EntListaModel>> ObtenerItemsCodigo(@PathVariable String codigo) {
        DataConfi();
        EntLista BS = new EntLista();
        ArrayList<EntListaModel> Items = new ArrayList<>();
        ArrayList<EntListaEntity> Data = BS.ObtenerCodigo(codigo);

        for (EntListaEntity Item : Data) {
            Items.add(new EntListaModel(Item));
        }

        return new ResponseAPI<ArrayList<EntListaModel>>(Items, true, "");
    }

}

File: SharedApi/Space/Space.Server/Controllers/EntListaController.cs
using Microsoft.AspNetCore.Mvc;
using Space.BusinessLayer;
using Space.EntityLayer;


namespace Space.Server.Controllers
{
    [Route("api/[controller]")]
    [ApiController]
    public class EntListaController : ControllerBase
    {

        Base d = new Base();
        [HttpGet]
        [Route("ObtenerItemsCodigo/{codigo}")]
        public ResponseAPI<List<EntListaModel>> ObtenerItemsCodigo(String codigo)
        {
            try
            {
                d.Configurar();
                List<EntListaModel> Lista = new List<EntListaModel>();

                var Items = EntLista.ObtenerCodigo(codigo);
                foreach (var Item in Items)
                {
                    Lista.Add(new EntListaModel
                    {
                        ListaId= Item.ListaId,
                        Nombre= Item.Nombre
                    });
                }

                return new ResponseAPI<List<EntListaModel>>(Lista, true);
            }
            catch (Exception ex)
            {
                return new ResponseAPI<List<EntListaModel>>(ex.Message.ToString());
            }
        }


      

    }
}
